src/Optimizers/MoveToComponents.ts

```diff
@@ -0,0 +1,79 @@
+import { OptimizerInterface } from '../Models/OptimizerInterface';
+import { ReportElement } from '../Models/Report';
+import { ComponentProvider } from '../ComponentProvider';
+import { AsyncAPIDocument } from '@asyncapi/parser';
+import { compareComponents } from '../Utils';
+
+export class MoveToComponents implements OptimizerInterface {
+  provider: ComponentProvider;
+
+  constructor(private document: AsyncAPIDocument) {
+    this.provider = ComponentProvider.getInstance(document);
+  }
+
+  //yet to be implemented.
+  getReport = (): ReportElement[] => {
+    return this.findDuplicateComponents(this.provider.schemas, 'schema').concat(
+      this.findDuplicateComponents(this.provider.messages, 'message'),
+      this.findDuplicateComponents(this.provider.parameters, 'parameter')
+    );
+  }
+  getMatch = (key: string, value: any, components: Map<string, any>): string => {
+    let matchedKey = '';
+    if (key.startsWith('components.')) {
```

Comment:
IMO you should have a utils function `isComponent` which takes a string as input and does this startsWith check. Using this function everywhere improves readability and in case the definition of `component` changes, you need to do it only at one place.